src/freertos_drivers/net_cc32xx/CC32xxWiFi.cxx

```diff
@@ -575,6 +577,17 @@ void CC32xxWiFi::run_on_network_thread(std::function<void()> callback)
 }
 
 /*
+ * CC32xxWiFi::add_http_get_callback()
+ */
+void CC32xxWiFi::add_http_get_callback(std::pair<std::function<std::string()>,
+                                                 const char *> callback)
+{
+    OSMutexLock l(&lock_);
+    httpGetCallbacks_.emplace_back(callback);
```

Comment:
write it like this:
....emplace_back(std::move(callback));
this will avoid a copy.

note that this is a good candidate function for inlining, but you have to put it into the header for that.